src/app/reset-password/page.tsx

```diff
@@ -0,0 +1,154 @@
+import Notify from '@config/notiflix-config';
+import { PasswordResetProvider } from '@/components/password-reset-provider';
+
+const SEND_RESET_PASSWORD_MUTATION = gql`
+  mutation SendResetPasswordToken($email: String!) {
+    sendResetPasswordToken(email: $email)
+  }
+`;
+
+const RESET_PASSWORD_MUTATION = gql`
+  mutation ResetPassword($password: String!, $token: String!) {
+    resetPassword(password: $password, token: $token)
+  }
+`;
+
+export default function Page() {
+  const searchParams = useSearchParams();
+  const token = searchParams.get("token");
+  const router = useRouter();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const toggleShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const onPasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPassword(event.target.value);
+  };
+
+  const onConfirmPasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setConfirmPassword(event.target.value);
+  };
+
+  const onEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(event.target.value);
+  };
+
+  const toggleLoader = (state: boolean) => {
+    const button: HTMLElement | null = document.getElementById("login-button");
+    if (button) {
+      button.classList.toggle("loading", state);
+    }
+  };
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (token) {
+      if (password && confirmPassword) {
+        if (password !== confirmPassword) {
+          Notify.failure("Passwords do not match!");
+        } else {
+          resetPassword();
+        }
+      } else {
+        Notify.failure("Please enter both password and confirmation password!");
+      }
+    } else if (email) {
+      sendResetPassword();
+    }
+  };
+
+  const [sendResetPassword, { loading: sendLoading, error: sendError }] = useMutation(SEND_RESET_PASSWORD_MUTATION, {
+    variables: {
+      email: email,
+    },
+    onCompleted: (data) => {
+      toggleLoader(false);
+      Notify.success(`Email sent!`);
+      router.push('/');
+    },
+    onError: (error) => {
+      toggleLoader(false);
+      Notify.failure(`${error.message}!`);
+    },
+  });
+
+  const [resetPassword, { loading: resetLoading, error: resetError }] = useMutation(RESET_PASSWORD_MUTATION, {
+    variables: {
+      password: confirmPassword,
+      token: token,
+    },
+    onCompleted: (data) => {
+      toggleLoader(false);
+      Notify.success(`Password Changed Successfully!`);
+      router.push('/login');
+    },
+    onError: (error) => {
+      toggleLoader(false);
+      console.log('onError error:', error);
+      Notify.failure(`${error.message}!`);
+    },
+  });
+
+  return (
+    <div className="flex flex-col h-screen">
```

Comment:
I'll create component for that. If it's done before merging it would be neat to adjust that part to reuse it. Same thing for the confirmation pr.